src/ngraph_builder.cc

```diff
@@ -3870,6 +3870,34 @@ static Status TranslateTransposeOp(
   TF_RETURN_IF_ERROR(
       GetStaticInputVector(op, 1, static_input_map, &permutation));
 
+  auto ng_input_rank = ng_input->get_shape().size();
+  std::unordered_set<int> axes;
+  for (auto p : permutation) {
+    if (p < ng_input_rank && !axes.count(p)) {
+      axes.insert(p);
+    } else {
+      auto missing_axes = 0;
+      for (int i = 0; i < ng_input_rank; i++) {
+        if (std::find(permutation.begin(), permutation.end(), i) ==
+            permutation.end()) {
+          missing_axes = i;
+          break;
+        }
+      }
+
+      std::ostringstream oss;
+      // Convert all but the last element to avoid a trailing ","
+      std::copy(permutation.begin(), permutation.end() - 1,
+                std::ostream_iterator<int>(oss, ", "));
+
+      // Now add the last element with no delimiter
+      oss << permutation.back();
```

Comment:
There is an `ngraph::join`: https://github.com/NervanaSystems/ngraph/blob/master/src/ngraph/util.hpp#L49
`ng::join(permutation)` should work